Cotton Classing Database File Import Plugin/Plugin.cs

```diff
@@ -0,0 +1,147 @@
+            {
+                //A plugin publisher can choose to create one or multiple application data models as appropriate for the data
+                ApplicationDataModel adm = new ApplicationDataModel();
+                adm.Catalog = new Catalog() { Description = $"ADAPT data transformation of Publisher data {DateTime.Now.ToShortDateString()} {dataPath}" };
+                //var loads = new List<AgGateway.ADAPT.ApplicationDataModel.LoggedData.Load>();
+                models.Add(adm);
+
+                foreach (string myDataFile in myDataFiles)
+                {
+                    //Import each file
+                    PublisherDataModel.Data data = new PublisherDataModel.Data();
+                    data.NDBData = PublisherDataModel.FlatFileHelper.ConvertFlatFileToModel(File.ReadAllText(myDataFile, System.Text.Encoding.Default));
+                    DataMappers.DataMapper.MapData(data, adm);
+                }
+            }
+
+            return models;           
+        }
+
+        /// <summary>
+        /// Export works just like import, except the Mappers should work in the reverse direction
+        /// </summary>
+        /// <param name="dataModel">The ADAPT adm to export to the plugin's format</param>
+        /// <param name="exportPath">Path to publish to</param>
+        /// <param name="properties">Any proprietary values the user may pass in customizing the export</param>
+        void IPlugin.Export(ApplicationDataModel dataModel, string exportPath, Properties properties)
+        {
+            throw new NotImplementedException();
+        }
+
+
+        /// <summary>
+        /// GetProperties is an optional feature to return 
+        /// basic key/value information about a data set without
+        /// completing the complete transformation
+        /// </summary>
+        /// <param name="dataPath"></param>
+        /// <returns></returns>
+        Properties IPlugin.GetProperties(string dataPath)
+        {
+            
+            throw new NotImplementedException();
+        }
+
+
+        /// <summary>
+        ///Initialize is an optional feature if a publisher wishes
+        ///to secure use of the plugin with specific arguments  
+        ///or otherwise customize the behavior with a particular set of parameters
+        /// <param name="args"></param>
+        void IPlugin.Initialize(string args)
+        {
+
+        }
+
+        /// <summary>
+        /// Determines whether the folder contains data that this plugin can import
+        /// </summary>
+        /// <param name="dataPath"></param>
+        /// <param name="properties"></param>
+        /// <returns></returns>
+        bool IPlugin.IsDataCardSupported(string dataPath, Properties properties)
+        {
+            //In this simple example, we are simply looking for the myjson extension to identify data in our format
+            if (Directory.GetFiles(dataPath, "*.txt", SearchOption.AllDirectories).Any())
```

Comment:
I believe we discussed changing the SearchOption to TopDirectoryOnly and providing guidance in the readme or elsewhere to point the plugin at the specific directory to avoid crawling down and eventually opening all text files in a path.